kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -900,6 +910,93 @@ public void testAddingS3ObjectTags() throws Exception{
     verifyTags(expectedTaggedFiles);
   }
 
+  @Test (expected = DataException.class)
+  public void testExceptionOnNullKeys() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, null,
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, sampleHeaders());
+    writeRecordWithKeysAndHeaders(faultyRecord);
+  }
+
+  @Test (expected = DataException.class)
```

Comment:
Please use `assertThrows`. It's the up-to-date way to assert on exceptions. And even expect other details related to the exception, besides their type.